### PR TITLE
Implemented initial hook with related Redux setup - Closes #4371

### LINKED ISSUE #4371: Create a custom hook for application management
### Description
A Custom hooks for the application is an interface to handle all the logic related to application management. This enables the component/hooks to use applications logic from one single place

### Acceptance Criteria
- Fetch list of applications, the list should be a mix of user applications and default application fetched from the server
- Enable ‘isPin’ flag for default applications from `useApplicationsPin` hooks then sort the list by pinned priority
- Enable memoization for applications
- Update status by fetching status from the server
- Allow users to add an application to the store by `setApplication` method
- Allow users to fetch a specific application by chainId via `getApplicationByChainId`
- Allow users to remove specific application by chainId from local storage via `deleteApplicationByChainId`
- If the current application is same as the selected application for remove, we need set Lisk as current application  via `useCurrentApplication`
- Add required unit tests
	
### Additional Information
- Suggested component name `useApplicationsManagement`
- Create reducer for blockchain application management and combine them as blockchainApplications 
`state.blockChainApplications.management`
```
function useApplicationManagement() {

  return {applications, setApplication, deleteApplicationByChainId, getApplicationByChainId};
}
```
blocked by #4373, #4357

### PR DESCRIPTION
**Pull request by ikem-legend**
### What was the problem?

This PR resolves #4371

### How was it solved?

- [x] Fetch list of applications
- [x] Enable ‘isPin’ flag for default applications from `useApplicationsPin` hooks then sort the list by pinned priority
- [x] Enable memoization for applications
- [x] Implement `setApplication` method to enable users to add an application to the store
- [x] Implement `getApplicationByChainId` to enable users to fetch a specific application by chainId
- [x] Implement `deleteApplicationByChainId` to enable users to remove specific application by chainId from local storage

### How was it tested?

Via unit tests

### CODE REVIEW

File: src/modules/blockchainApplication/manage/store/action.js [hunk 1]
```diff
@@ -10,6 +10,26 @@ export const toggleApplicationPin = (chainId) => ({
...
+
+/**
+ * Trigger this action to delete blockchain application
+ *
+ * @returns {Object} - Action object
+ */
+export const deleteApplication = (chainId) => ({
+  type: actionTypes.deleteApplicationByChainId,
+  data: chainId,
+});
```
Comment (soroushm) [inline src/modules/blockchainApplication/manage/store/action.js]: a bit wired to me passing data here shall we pass chainId
```suggestion
export const deleteApplication = (chainId) => ({
  type: actionTypes.deleteApplicationByChainId,
  chainId,
});
```

File: src/modules/blockchainApplication/manage/store/reducer.js [hunk 2]
```diff
@@ -24,7 +35,34 @@ export const pins = (state = [], { type, chainId }) => {
...
+        state[data.chainID].nodeURL.push(data.nodeURL);
+      } else {
+        state[data.chainID] = data;
+      }
+      return state;
+
+    case actionTypes.deleteApplicationByChainId: {
+      delete state[data];
+      return { ...state };
+    }
```
Comment (soroushm) [inline src/modules/blockchainApplication/manage/store/reducer.js]: improve naming please data is a bit wired for me
```suggestion
export const applications = (state = initialState.applications, { type, application, chainId }) => {
  switch (type) {
    case actionTypes.addApplicationByChainId:
      // In cases where a new node for an existing application is being added,
      // the new node url should be appended to the nodeURL array of the application
      if (application.chainID in state) {
        state[application.chainID].nodeURL.push(data.nodeURL);
      } else {
        state[application.chainID] = data;
      }
      return state;

    case actionTypes.deleteApplicationByChainId: {
      delete state[chainId];
      return { ...state };
    }
```

**Review by soroushm — DISMISSED** [review]

File: tests/fixtures/blockchainApplicationsManage.js [hunk 3]
```diff
@@ -1,12 +1,60 @@
...
-    logoURL: 'https://service.lisk.com',
+    serviceURL: ['https://service.lisk.com'],
     lastUpdated: 123456789,
+    isDefault: true,
+  },
+  {
+    name: 'Colecti',
+    chainID: 'mi34vyyd12g2lkf0rza1irws',
+    state: 'active',
+    serviceURL: ['https://service.colecti.com'],
```
Comment (eniolam1000752) [inline tests/fixtures/blockchainApplicationsManage.js]: this should be `serviceURL`

**Review by eniolam1000752 — CHANGES_REQUESTED** [review]

**Review by soroushm — APPROVED** [review]

**Review by eniolam1000752 — APPROVED** [review]
LGTM

**Pull request merged**